### PR TITLE
Ingestion UX study changes

### PR DESCRIPTION
**Pull request by rakshith91**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/monitor/azure-monitor-ingestion/azure/monitor/ingestion/_models.py [hunk 1]
```diff
@@ -7,21 +7,29 @@
...
 
 class UploadLogsStatus(with_metaclass(CaseInsensitiveEnumMeta, str, Enum)):
     SUCCESS = "Success"
     PARTIAL_FAILURE = "PartialFailure"
+    FAILURE = "Failure"
 
+class UploadLogsError:
+    """The errors for any failed logs
+    """
+    def __init__(self, **kwargs) -> None:
```
Comment (lmazuel) [inline sdk/monitor/azure-monitor-ingestion/azure/monitor/ingestion/_models.py:22]: Should this be:
```python
def __init__(self, *, failed_logs=None, error=None) -> None:
```

File: sdk/monitor/azure-monitor-ingestion/azure/monitor/ingestion/_operations/_patch.py [hunk 2]
```diff
@@ -57,25 +64,37 @@ def upload( # pylint: disable=arguments-renamed, arguments-differ
...
+                        content_encoding="gzip",
+                        **kwargs
+                    )
+                except Exception as err: # pylint: disable=bare-exception
+                    results.append(UploadLogsError(
+                        error = err,
+                        failed_logs = request
+                    ))
+        
+        if len(results) == 0:
```
Comment (lmazuel) [inline sdk/monitor/azure-monitor-ingestion/azure/monitor/ingestion/_operations/_patch.py:91]:
```suggestion
        if not results:
```

File: sdk/monitor/azure-monitor-ingestion/azure/monitor/ingestion/aio/_operations/_patch.py [hunk 3]
```diff
@@ -25,26 +39,73 @@ async def upload( # pylint: disable=arguments-renamed, arguments-differ
         :param stream: The streamDeclaration name as defined in the Data Collection Rule.
         :type stream: str
         :param body: An array of objects matching the schema defined by the provided stream.
-        :type body: list[any]
+        :type body: list[Any] or IO
```
Comment (lmazuel) [inline sdk/monitor/azure-monitor-ingestion/azure/monitor/ingestion/aio/_operations/_patch.py:42]: Inconsistent with signature. "logs" or "body"?

File: sdk/monitor/azure-monitor-ingestion/azure/monitor/ingestion/_models.py (hunk 1 above)
Comment (rakshith91) [reply sdk/monitor/azure-monitor-ingestion/azure/monitor/ingestion/_models.py:22]: These are ivars - I don't see why we need to make them positional

File: sdk/monitor/azure-monitor-ingestion/azure/monitor/ingestion/aio/_operations/_patch.py [hunk 4]
```diff
@@ -24,27 +38,74 @@ async def upload( # pylint: disable=arguments-renamed, arguments-differ
         :type rule_id: str
         :param stream: The streamDeclaration name as defined in the Data Collection Rule.
         :type stream: str
-        :param body: An array of objects matching the schema defined by the provided stream.
-        :type body: list[any]
+        :param logs: An array of objects matching the schema defined by the provided stream.
+        :type logs: list[JSON] or IO
+        :keyword max_concurrency: Number of parallel threads to use when logs size is > 1mb.
```
Comment (scottaddie) [inline sdk/monitor/azure-monitor-ingestion/azure/monitor/ingestion/aio/_operations/_patch.py:43]: Is that 1 MB limit enforced by the service? If so, let's avoid saying "is > 1mb" and instead say something like "exceeds the limit imposed by the service".

Comment (rakshith91) [reply sdk/monitor/azure-monitor-ingestion/azure/monitor/ingestion/aio/_operations/_patch.py:43]: that's a good idea - yes it's imposed by the service

**Review by pvaneck — APPROVED** [review]

**Pull request merged**